src/app/services/app-state.service.ts

```diff
@@ -77,11 +77,16 @@ export class AppStateService {
 
 
   setRecordSelected(cinchyId: number | null, doNotReloadForm: boolean = false): void {
-
     this._rowId = cinchyId;
-
     this.onRecordSelected$.next({ cinchyId, doNotReloadForm });
 
+    if (cinchyId == null) {
```

Comment:
This should be `===`. Also, since you've now set `this._rowId`, it should be used for the comparison. This doesn't happen asynchronously, so there's no concern about the value changing elsewhere.

Reply:
done